XiEditor/EditViewController.swift

```diff
@@ -155,6 +155,14 @@ class EditViewController: NSViewController, EditViewDataSource {
     
     // MARK: - System Events
     override func keyDown(with theEvent: NSEvent) {
+        if (NSApplication.shared().delegate as! AppDelegate).vimMode {
+            if let codepoint = theEvent.characters?.unicodeScalars.first?.value {
+                if codepoint > 0 && codepoint < 128 {
```

Comment:
Is this just checking for 7-bit ascii?  If so, is there any link to documentation of where these magic values come from?  Also, would be helpful to understand why only the first key is being checked.